cypress/e2e/pages/topicPage/tests.js

```diff
@@ -31,69 +31,73 @@ export default ({ service, pageType, variant }) => {
-        cy.getPageData({
-          service,
-          pageType: 'topic',
-          variant,
-        }).then(({ body }) => {
-          topicTitle = body.data.title;
-          variantTopicId = body.data.variantTopicId;
-          pageCount = body.data.pageCount;
-          numberOfItems = body.data.curations[0].summaries.length;
-          firstItemHeadline = body.data.curations[0].summaries[0].title;
-          messageBanner = body.data.curations.find(
-            curation =>
-              curation.visualProminence === 'NORMAL' &&
-              curation.visualStyle === 'BANNER',
-          );
-        });
+        const data = await cy.getPageDataFromWindow();
+        const { pageData } = data;
+        topicTitle = pageData.title;
+        variantTopicId = pageData.scriptSwitchId;
+        pageCount = pageData.pageCount;
+        numberOfItems = pageData.curations?.[0]?.summaries.length;
+        firstItemHeadline = pageData.curations?.[0]?.summaries?.[0]?.title;
+        messageBanner = pageData.curations?.find(
+          curation =>
+            curation.visualProminence === 'NORMAL' &&
+            curation.visualStyle === 'BANNER',
+        );
+
         cy.log(`topic id ${topicId}`);
       }
     });
 
     describe(`Page content`, () => {
+      beforeEach(() => {
+        // make sure we always start from the path being tested to make the tests deterministic and not reliant on order
+        // as otherwise some tests can change the path and affect subsequent tests (i.e. when you change page script)
+        cy.visit(currentPath);
+      });
       it('should render a H1, which contains/displays topic title', () => {
         cy.log(Cypress.env('currentPath'));
-
         cy.get('h1').should('contain', topicTitle);
       });
+
       it('should render the correct number of items', () => {
         // Print SIMORGH_DATA if the number of promos on the page does not match
-        // the number of promos in the data from the BFF
+        // the number of promos in the data from window.SIMORGH_DATA
         // This is to help find out why sometimes a promo doesn't show on the page
-        cy.log(`Number of promos in BFF data${numberOfItems}`);
+        cy.log(`Number of promos in SIMORGH data${numberOfItems}`);
         const selector = '[data-testid="topic-promos"]:first > li';
         const promoCount = Cypress.$(selector).length;
         cy.log(`Number of promos on the page${promoCount}`);
-
         if (promoCount !== numberOfItems) {
           cy.window().then(win => {
             const pageData = win.SIMORGH_DATA;
-
             cy.log(pageData);
           });
         }
-
         // Checks number of items on page
         cy.get('[data-testid="topic-promos"]')
           .first()
           .children()
           .its('length')
           .should('eq', numberOfItems);
       });
-      it.skip('First item has correct headline', () => {
+
+      it('First item has correct headline', () => {
```

Comment:
Let's keep a watch on the simorgh-alerts channel for failures on this one. The reason I skipped it originally was because sometimes the promo headline is different from the full article headline, which causes a failure, but it is expected as sometimes there is a shortHeadline which is different from the longer one used in the article.

I am thinking maybe we either remove this test or just leave it to see if this still happens. What do you think @karinathomasbbc ?

Reply:
I think that's a good shout - add logging if it doesn't exist, and keep an eye on the channel so that we can quickly disable / fix if it becomes an issue?

Reply:
Yes, that makes sense .. I already added a `cy.log`, let me know if that's enough to monitor the test

Reply:
Ah yes - in fact, the test won't fail anymore because you're checking if `firstItemHeadline` has a value and then exiting, so the tests won't be "flaky" anymore, but there is no guarantee that it will always run the assertions.